internal/services/cdn/cdn_frontdoor_origin_resource.go

```diff
@@ -0,0 +1,269 @@
+package cdn
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/parse"
+	track1 "github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/sdk/2021-06-01"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceCdnFrontdoorOrigin() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceCdnFrontdoorOriginCreate,
+		Read:   resourceCdnFrontdoorOriginRead,
+		Update: resourceCdnFrontdoorOriginUpdate,
+		Delete: resourceCdnFrontdoorOriginDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.FrontdoorOriginID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"cdn_frontdoor_origin_group_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FrontdoorOriginGroupID,
+			},
+
+			// HostName cannot be null or empty.;
+			"host_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"cdn_frontdoor_origin_id": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+			},
+
+			"enable_health_probes": {
```

Comment:
`enable_health_probes` is a bit confusing given that it translates to:

![image](https://user-images.githubusercontent.com/17407022/161082655-b1c886fd-68d0-4a36-8ac9-eae338e66159.png)

and

![image](https://user-images.githubusercontent.com/17407022/161082751-016adc25-e45b-43d5-825d-f7bb2f8e6ec2.png)

would just "enabled" not be the better name here?

Reply:
@heoelri potentially... I will have to look at the schema again once I finally get everything working together. When I was naming the fields I was going from the swagger, so in the resource this might be true.

Reply:
@heoelri This is the result of a confusing UI, the **Enable this origin** value is coming from the Endpoint **enabled** field and the **Status** under the **Origin Host Name** is coming from the **enable_health_probes** value. I do think I will flip that around to make it **health_probes_enabled**, that is more inline with the new coding standards. Thanks for your comment. 🚀